src/main/resources/api/data-repository-openapi.yaml

```
@@ -6711,6 +6763,101 @@ components:
           items:
             type: string
 
+    SnapshotBuilderAccessRequest:
+      type: object
+      properties:
+        name:
+          type: string
+        researchPurposeStatement:
+          type: string
+        datasetRequest:
+          $ref: '#/components/schemas/SnapshotBuilderRequest'
+      description: Request for access to a snapshot.
+
+    SnapshotBuilderRequest:
+      type: object
+      properties:
+        cohorts:
+          type: array
+          items:
+            $ref: '#/components/schemas/SnapshotBuilderCohort'
+        conceptSets:
+          type: array
+          items:
+            $ref: '#/components/schemas/SnapshotBuilderDatasetConceptSets'
```

I think this type should be called `SnapshotBuilderDatasetConceptSet` since it's a single concept set. It's an existing type but it'd be good to rename it now if it's not a lot of extra work.
```suggestion
            $ref: '#/components/schemas/SnapshotBuilderDatasetConceptSet'
```